File: java/cdk8s-plus-elasticsearch-query/src/main/java/imports/co/elastic/k8s/kibana/KibanaSpecHttpServiceSpecPortsTargetPort.java
package imports.co.elastic.k8s.kibana;

/**
 * Number or name of the port to access on the pods targeted by the service.
 * <p>
 * Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. If this is a string, it will be looked up as a named port in the target Pod's container ports. If this is not specified, the value of the 'port' field is used (an identity map). This field is ignored for services with clusterIP=None, and should be omitted or set equal to the 'port' field. More info: https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.79.0 (build b22f628)", date = "2023-03-31T16:02:16.005Z")
@software.amazon.jsii.Jsii(module = imports.co.elastic.k8s.kibana.$Module.class, fqn = "coelastick8skibana.KibanaSpecHttpServiceSpecPortsTargetPort")
public class KibanaSpecHttpServiceSpecPortsTargetPort extends software.amazon.jsii.JsiiObject {

    protected KibanaSpecHttpServiceSpecPortsTargetPort(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected KibanaSpecHttpServiceSpecPortsTargetPort(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    public static @org.jetbrains.annotations.NotNull imports.co.elastic.k8s.kibana.KibanaSpecHttpServiceSpecPortsTargetPort fromNumber(final @org.jetbrains.annotations.NotNull java.lang.Number value) {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(imports.co.elastic.k8s.kibana.KibanaSpecHttpServiceSpecPortsTargetPort.class, "fromNumber", software.amazon.jsii.NativeType.forClass(imports.co.elastic.k8s.kibana.KibanaSpecHttpServiceSpecPortsTargetPort.class), new Object[] { java.util.Objects.requireNonNull(value, "value is required") });
    }

    public static @org.jetbrains.annotations.NotNull imports.co.elastic.k8s.kibana.KibanaSpecHttpServiceSpecPortsTargetPort fromString(final @org.jetbrains.annotations.NotNull java.lang.String value) {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(imports.co.elastic.k8s.kibana.KibanaSpecHttpServiceSpecPortsTargetPort.class, "fromString", software.amazon.jsii.NativeType.forClass(imports.co.elastic.k8s.kibana.KibanaSpecHttpServiceSpecPortsTargetPort.class), new Object[] { java.util.Objects.requireNonNull(value, "value is required") });
    }

    public @org.jetbrains.annotations.NotNull java.lang.Object getValue() {
        return software.amazon.jsii.Kernel.get(this, "value", software.amazon.jsii.NativeType.forClass(java.lang.Object.class));
    }
}

File: go/cdk8s-plus-elasticsearch-query/imports/kibanak8selasticco/KibanaSpecHttpServiceSpecPortsTargetPort.go
package kibanak8selasticco

import (
	_init_ "example.com/cdk8s-plus-elasticsearch-query/imports/kibanak8selasticco/jsii"
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"
)

// Number or name of the port to access on the pods targeted by the service.
//
// Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. If this is a string, it will be looked up as a named port in the target Pod's container ports. If this is not specified, the value of the 'port' field is used (an identity map). This field is ignored for services with clusterIP=None, and should be omitted or set equal to the 'port' field. More info: https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
type KibanaSpecHttpServiceSpecPortsTargetPort interface {
	Value() interface{}
}

// The jsii proxy struct for KibanaSpecHttpServiceSpecPortsTargetPort
type jsiiProxy_KibanaSpecHttpServiceSpecPortsTargetPort struct {
	_ byte // padding
}

func (j *jsiiProxy_KibanaSpecHttpServiceSpecPortsTargetPort) Value() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"value",
		&returns,
	)
	return returns
}


func KibanaSpecHttpServiceSpecPortsTargetPort_FromNumber(value *float64) KibanaSpecHttpServiceSpecPortsTargetPort {
	_init_.Initialize()

	if err := validateKibanaSpecHttpServiceSpecPortsTargetPort_FromNumberParameters(value); err != nil {
		panic(err)
	}
	var returns KibanaSpecHttpServiceSpecPortsTargetPort

	_jsii_.StaticInvoke(
		"kibanak8selasticco.KibanaSpecHttpServiceSpecPortsTargetPort",
		"fromNumber",
		[]interface{}{value},
		&returns,
	)

	return returns
}

func KibanaSpecHttpServiceSpecPortsTargetPort_FromString(value *string) KibanaSpecHttpServiceSpecPortsTargetPort {
	_init_.Initialize()

	if err := validateKibanaSpecHttpServiceSpecPortsTargetPort_FromStringParameters(value); err != nil {
		panic(err)
	}
	var returns KibanaSpecHttpServiceSpecPortsTargetPort

	_jsii_.StaticInvoke(
		"kibanak8selasticco.KibanaSpecHttpServiceSpecPortsTargetPort",
		"fromString",
		[]interface{}{value},
		&returns,
	)

	return returns
}

